frontend/tests/jest/tests/ProductSection.test.tsx

```diff
@@ -268,6 +268,66 @@ describe('ProductSection Tests', () => {
       });
    });
 
+   describe('Prop 65 Warning Tests', () => {
+      test('renders prop 65 warning', async () => {
+         const battery = getProductOfType('battery');
+
+         renderWithAppContext(
+            <ProductSection
+               product={battery}
+               selectedVariant={battery.variants[0]}
+               onVariantChange={jest.fn()}
+               internationalBuyBox={null}
+            />
+         );
+         const prop65WarningText = await screen.findByText(/prop 65 warning/i);
+         (expect(prop65WarningText) as any).toBeVisible();
+      });
+
+      test('does not render prop 65 warning', async () => {
+         const tool = getProductOfType('tool');
+
+         renderWithAppContext(
+            <ProductSection
+               product={tool}
+               selectedVariant={tool.variants[0]}
+               onVariantChange={jest.fn()}
+               internationalBuyBox={null}
+            />
+         );
+         (
+            expect(screen.queryByText(/prop 65 warning/i)) as any
+         ).not.toBeInTheDocument();
+      });
+
+      test('renders prop 65 info popup', async () => {
+         const battery = getProductOfType('battery');
+
+         renderWithAppContext(
+            <ProductSection
+               product={battery}
+               selectedVariant={battery.variants[0]}
+               onVariantChange={jest.fn()}
+               internationalBuyBox={null}
+            />
+         );
+
+         act(() => {
+            screen.getByLabelText('read more about the warning').click();
+         });
```

Comment:
TIL about `act()`. It looks like we can use `fireEvent` instead of `act()`, [ref](https://testing-library.com/docs/preact-testing-library/api/#fireevent).
```suggestion
         fireEvent.click(screen.getByLabelText('read more about the warning'));
```